GameServerLib/Logic/GameObjects/Stats/ChampionStats.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LeagueSandbox.GameServer.Logic.GameObjects.Stats
+{
+    [StructLayout(LayoutKind.Explicit)]
+    public class ChampionStats
```

Comment:
Use properties, don't expose public fields

Reply:
From what I checked there is a problem with applying FieldOffsetAtrribute to properties (see here: https://stackoverflow.com/a/28505211) - I can try to check if that's work but I dont think so.

The solution for that will be to write some function that serializes the data, but that's sort of redundancy because all we want is to write the variables in their offset.

Reply:
If you convert this to a struct it should be fine to have non-properties. Is there anything preventing this from being a struct?

Reply:
No, but then I think I should write serializer class for the GetBytes method, or I may I keep it in the struct?

Reply:
Feel free to keep it in the struct as far as I'm concerned

Reply:
There is a problem after changing the class to struct, everytime you want to change variable in the struct you need to create a new struct (see here: https://stackoverflow.com/questions/1747654/cannot-modify-the-return-value-error-c-sharp), that's beacuse the struct is a value type not like class.

Reply:
Can you serialize private variables? If so, the "proper" way to go about this would be
```CS
[FieldOffset(0)]
private int _assists = 0;
public int Assists { get { return _assists; } set { _assists = value; } }
```
however, that would be fairly redundant.

I think it's fine to deviate from our normal style guide in this case unless someone else has a better suggestion. Just make sure to comment in the beginning of the class that it's exempt from the style guide.

Reply:
I like this serializing solution. Looks redundant but that's exactly how autoprops work and it'd provide proper encapsulation. So looks good to me

Reply:
So what to implement?
the private variable+public property solution?
or keep it as is?

Reply:
Sorry for the confusion 😄, I meant that it's fine to keep it as it is (which deviates from our styleguide), just add a commet mentioning that it's been discussed and it's fine to deviate from the style guide (you can link this PR for example).